### PR TITLE
Extract static `DATA_KEY` & `EVENT_KEY` to base-component

### PR DESCRIPTION
**Pull request by GeoSot**
* Force each plugin that extends base-components to implement a static method `NAME()`
* Remove redundant `NAME` argument from 'Utils.defineJQueryPlugin' & fix test

### CODE REVIEW

File: js/src/util/index.js [hunk 1]
```diff
@@ -214,16 +214,16 @@ const onDOMContentLoaded = callback => {
...
   onDOMContentLoaded(() => {
     const $ = getjQuery()
     /* istanbul ignore if */
     if ($) {
-      const JQUERY_NO_CONFLICT = $.fn[name]
-      $.fn[name] = plugin.jQueryInterface
-      $.fn[name].Constructor = plugin
-      $.fn[name].noConflict = () => {
-        $.fn[name] = JQUERY_NO_CONFLICT
+      const JQUERY_NO_CONFLICT = $.fn[plugin.NAME]
```
Comment (alecpl) [inline js/src/util/index.js:222]: I'm not sure, but doing `const name = plugin.Name` first should make this code shorter and the change simpler.

File: js/src/base-component.js [hunk 2]
```diff
@@ -43,6 +43,14 @@ class BaseComponent {
   static get VERSION() {
     return VERSION
   }
+
+  static get NAME() {
+    throw new Error('You have to implement the static method "NAME", for each component!')
+  }
+
+  static get DATA_KEY() {
+    return `bs.${this.NAME}`
```
Comment (alecpl) [inline js/src/base-component.js:52]: Wouldn't just `'bs.' + this.Name` be cleaner and shorter?

**Review by alecpl — COMMENTED** [review]
I love your refactorings @GeoSot

File: js/src/base-component.js (hunk 2 above)
Comment (GeoSot) [reply js/src/base-component.js:52]: Thank you for our review @alecpl 
I think, we avoid string concatenation, using `+` , so I am not sure which is the 'best'

**Review by alpadev — APPROVED** [review]

**Pull request merged**